internal/pkg/api/server/buffer/service/service.go

```diff
@@ -37,51 +40,115 @@ func (*service) HealthCheck(dependencies.ForPublicRequest) (res string, err erro
 }
 
 func (*service) CreateReceiver(d dependencies.ForProjectRequest, payload *buffer.CreateReceiverPayload) (res *buffer.Receiver, err error) {
-	ctx := d.RequestCtx()
+	ctx, store := d.RequestCtx(), d.ConfigStore()
 
-	store := d.ConfigStore()
+	// Check if receiver limit is reached
+	count, err := store.CountReceivers(ctx, d.ProjectID())
+	if err != nil {
+		return nil, err
+	}
+	if count >= 100 {
+		return nil, &GenericError{
+			StatusCode: http.StatusUnprocessableEntity,
+			Name:       "buffer.resourceLimitReached",
+			Message:    "Maximum number of receivers per project is 100.",
+		}
+	}
```

Comment:
- `100` by mala byt nejaka konstanta
- a tento kod by mal byt v `store`
- v `service` iba prepajame nejaky interny kod s API payloads

Reply:
Ze `store` můžu vracet `GenericError`? Já jsem to dal sem kvůli tomu, že potřebuju specifikovat ten status kód `422`. Konstantu dodělám

Reply:
`store` by mal vratit nejaku konkretnu chybu, ktoru tu porovnas podla `error.Is` (hodnoty), alebo typu `error.As` a namapujes to na HTTP chybu.

Ak chyba nezodpoveda ziadnej HTTP chybe, tak ju iba vratis, a server vyhodi 500, a chybu zaloguje.

Reply:
Priklad z `templates` service:
https://github.com/keboola/keboola-as-code/blob/ff9db26d1318a221171387d3cd5525f8a0373a11/internal/pkg/api/server/templates/service/service.go#L491-L503

Reply:
Takto su jednotlive vrstvy oddelene a kazda ma svoju zodpovednost.
Zaroven je to aj dobre testovatelne a mozes jednoducho vymenit HTTP aj za nejaky iny protokol.